File: frontend/src/app/core/models/customer.ts
import { Dictionary } from "./dictionary";
import { Model } from "./model";

export class Customer extends Model{
    name!: string;
    lastname1!: string;
    lastname2!: string;
    fullname!: string;
    document_type!: Dictionary;
    document_number!: string;

    images: string[] = [];
}

File: backend/app/models/customer.rb
class Customer < ApplicationRecord
    belongs_to :document_type, class_name: 'Dictionary', foreign_key: 'document_type_id'

    has_many_attached :images
    validates :images, content_type: [:png, :jpg, :jpeg, :jfif]

    def json = {
        id: id,
        name: name,
        lastname1: lastname1,
        lastname2: lastname2,
        document_type: document_type.json,
        document_number: document_number,
    }

    def fullname = "#{name} #{lastname1} #{lastname2}"
end
